src/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -479,44 +516,49 @@ private static uint Div128By96(uint[] rgulNum, uint[] rgulDen)
              *   None.
              *
              ***********************************************************************/
-            private static uint IncreaseScale(uint[] rgulNum, uint ulPwr)
+            private static uint IncreaseScale(ref Buf12 bufNum, uint ulPwr)
             {
-                Split64 sdlTmp = new Split64();
+                ulong tmp = UInt32x32To64(bufNum.U0, ulPwr);
+                bufNum.U0 = (uint)tmp;
+                tmp >>= 32;
+                tmp += UInt32x32To64(bufNum.U1, ulPwr);
+                bufNum.U1 = (uint)tmp;
+                tmp >>= 32;
+                tmp += UInt32x32To64(bufNum.U2, ulPwr);
+                bufNum.U2 = (uint)tmp;
+                return (uint)(tmp >> 32);
+            }
 
-                sdlTmp.int64 = UInt32x32To64(rgulNum[0], ulPwr);
-                rgulNum[0] = sdlTmp.Low32;
-                sdlTmp.int64 = UInt32x32To64(rgulNum[1], ulPwr) + sdlTmp.High32;
-                rgulNum[1] = sdlTmp.Low32;
-                sdlTmp.int64 = UInt32x32To64(rgulNum[2], ulPwr) + sdlTmp.High32;
-                rgulNum[2] = sdlTmp.Low32;
-                return sdlTmp.High32;
+            private static void IncreaseScale64(ref Buf12 bufNum, uint ulPwr)
+            {
+                ulong tmp = UInt32x32To64(bufNum.U0, ulPwr);
+                bufNum.U0 = (uint)tmp;
+                tmp >>= 32;
+                tmp += UInt32x32To64(bufNum.U1, ulPwr);
+                bufNum.High64 = tmp;
             }
 
             /***
             * ScaleResult
             *
             * Entry:
-            *   rgulRes - Array of ULONGs with value, least-significant first.
-            *   iHiRes  - Index of last non-zero value in rgulRes.
+            *   bufRes - Array of ULONGs with value, least-significant first.
+            *   iHiRes  - Index of last non-zero value in bufRes.
             *   iScale  - Scale factor for this value, range 0 - 2 * DEC_SCALE_MAX
             *
             * Purpose:
             *   See if we need to scale the result to fit it in 96 bits.
             *   Perform needed scaling.  Adjust scale factor accordingly.
             *
             * Exit:
-            *   rgulRes updated in place, always 3 ULONGs.
-            *   New scale factor returned, -1 if overflow error.
+            *   bufRes updated in place, always 3 ULONGs.
+            *   New scale factor returned.
             *
             ***********************************************************************/
-            private static int ScaleResult(uint[] rgulRes, int iHiRes, int iScale)
+            private static int ScaleResult(Buf24* bufRes, uint iHiRes, int iScale)
```

Comment:
Can this be `ref Buf24` ?

Reply:
`ref` would require using `fixed` to get the pointer below.